locator/tablets.cc

```diff
@@ -878,10 +930,65 @@ std::unordered_set<sstring> tablet_aware_replication_strategy::recognized_tablet
     return opts;
 }
 
+namespace {
+
+// Create a simple map [data center -> index]. This index is used as position index in the small vector
+// in the tablet_to_dc_replication_factor_map.
+[[nodiscard]] absl::flat_hash_map<sstring, tablet_effective_replication_map::replication_factor_index_t> build_datacenter_map(const auto& data_centers_list) {
+    uint32_t datacenter_index = 0;
+    absl::flat_hash_map<sstring, tablet_effective_replication_map::replication_factor_index_t> datacenter_map;
+    datacenter_map.reserve(data_centers_list.size());
+    for (const sstring& datacenter : data_centers_list) {
+        datacenter_map.emplace(datacenter, datacenter_index);
+        ++datacenter_index;
+    }
+    return datacenter_map;
+}
+
+// Populate the `tablet_to_dc_replication_factor_map` with replication factors for each data center.
+// For each tablet index in the tablet map, it retrieves the current replica set using `get_endpoints_for_reading`
+// and counts replicas per data center.  Each data center's replication factor (RF) is saved under
+// its index according to `_datacenter_map`.
```

Comment:
Comment needs updating, it doesn't populate tablet_to_dc_replication_factor_map.